File: android/src/main/java/com/rnmqtt/models/Protocol.kt
package com.rnmqtt.models

enum class Protocol {
  WS {
    override fun urlPrefix(): String = "ws://"
     },
  WSS {
    override fun urlPrefix(): String = "wss://"
      },
  SSL {
    override fun urlPrefix(): String = "ssl://"
      },
  TCP {
    override fun urlPrefix(): String = "tcp://"
  };

  abstract fun urlPrefix(): String
}

File: ios/Protocol.swift
enum Protocol: String {
  case WS = "WS"
  case WSS = "WSS"
  case TCP_TLS = "TCP_TLS"
  case TCP = "TCP"

  func urlPrefix() -> String {
      switch self {
        case .WS:
          return "ws://"
        case .WSS:
          return "wss://"
        case .TCP_TLS:
          return "ssl://"
        default:
          return "tcp://"
      }
    }
}
